src/mono/wasm/runtime/binding_support.js

```diff
@@ -652,6 +695,368 @@ var BindingSupportLib = {
 			return js_obj;
 		},
 
+		_create_named_function: function (name, argumentNames, body, closure) {
+			var result = null, keys = null, closureArgumentList = null, closureArgumentNames = null;
+
+			if (closure) {
+				closureArgumentNames = Object.keys (closure);
+				closureArgumentList = new Array (closureArgumentNames.length);
+				for (var i = 0, l = closureArgumentNames.length; i < l; i++)
+					closureArgumentList[i] = closure[closureArgumentNames[i]];
+			}
+
+			var constructor = this._create_rebindable_named_function (name, argumentNames, body, closureArgumentNames);
+			result = constructor.apply (null, closureArgumentList);
+
+			return result;
+		},
+
+		_create_rebindable_named_function: function (name, argumentNames, body, closureArgNames) {
+			var strictPrefix = "\"use strict\";\r\n";
+			var uriPrefix = "", escapedFunctionIdentifier = "";
+
+			if (name) {
+				uriPrefix = "//# sourceURL=https://mono-wasm.invalid/" + name + "\r\n";
```

Comment:
Shouldn't this be more like .generated instead of .invalid?

Reply:
.invalid is the only reserved TLD that is guaranteed not to ever resolve to anything.

Reply:
I don't think it actually needs a scheme and a host so you could just use the name.